columnflow/tasks/framework/base.py

```diff
@@ -219,6 +222,36 @@ def _get_cfg_versions_dict(cls):
 
         return cls._cfg_versions_dict
 
+    @classmethod
+    def _get_cfg_resources_dict(cls):
+        if cls._cfg_resources_dict is None and law.config.has_section("resources"):
+            # helper to split resource values into key-value pairs themselves
+            def parse(key: str, value: str) -> tuple[str, list[tuple[str, Any]]]:
+                params = []
+                for part in value.split(","):
+                    part = part.strip()
+                    if not part:
+                        continue
+                    if "=" not in part:
+                        logger.warning_once(
+                            f"invalid_resource_{key}",
+                            f"resource for key {key} contains invalid instruction {part}, skipping",
+                        )
+                        continue
+                    param, value = (s.strip() for s in part.split("=", 1))
+                    params.append((param, value))
+                return key, params
```

Comment:
this parse function is rather general (parsing into dictionary by splitting at "=" and ",") and independent of this particular function/class. Might this be reused at other points? Might it be better to separate it from the class?

Reply:
So far, there is no other place where this kind of parsing is used, but I agree we should move it once this approach can be used more generally.